### ISSUE TITLE
authorize_action_for: wrong number of arguments when using Sequel gem

### ISSUE THREAD

**Issue by sporto** [nathanl/authority#100]
I am trying Authority with the sequel gem, this is my setup:

models/Enquiry.rb

```ruby
class Enquiry < Sequel::Model
	include Authority::Abilities

	...
end
```

controllers/enquiries_controller.rb

```ruby
class EnquiriesController < BaseController
	authorize_actions_for Enquiry
	
	def index
		render json: ...
	end
end
```

I get this error:

```
ArgumentError (wrong number of arguments (10 for 3..4)):
  authority (3.0.0) lib/authority.rb:36:in `enforce'
  authority (3.0.0) lib/authority/controller.rb:119:in `authorize_action_for'
  authority (3.0.0) lib/authority/controller.rb:136:in `run_authorization_check'
```

Any ideas?
Thanks

**Comment by nathanl** [comment]
Hmmm. I wonder what `*Enquiry` produces on [this line](https://code.google.com/p/picoc/).

Try this: `bundle open authority` and `puts` all the arguments about to be passed to `enforce` [here](https://github.com/nathanl/authority/blob/v3.0.0/lib/authority/controller.rb#L119)

**Comment by sporto** [comment]
This is what I found

```
    def run_authorization_check
      Rails.logger.debug '- instance_authority_resource'
      Rails.logger.debug instance_authority_resource.inspect # Enquiry
      authorize_action_for(*instance_authority_resource)
    end
```

In `run_authorization_check` instance_authority_resource == Enquiry sequel model

```
    def authorize_action_for(authority_resource, *options)
      Rails.logger.debug 'authorize_action_for'
      Rails.logger.debug authority_resource.inspect # <Enquiry @values={:id=>1, :state=>nil, :suburb=>nil, :postcode=>nil, :person_1_id=>1, :person_2_id=>2, :created_at=>2015-05-29 14:23:54 +1000, :updated_at=>2015-05-29 14:24:31 +1000}>

     ...
    end
```

But at this point `authority_resource` has become the first record in the db. So then it just passes all the record values as options.

I don't really understand how the *instance_authority_resource is making sequel grab the first record.

**Comment by sporto** [comment]
This snippet shows the same behaviour:

```
	task :sequel => :environment do
		e = Enquiry

		def foo(res, *options)
			puts res.inspect #<Enquiry:0x007fd16b3bb9d0>
		end

		puts e.inspect # Enquiry
		foo(*e)
	end
```

**Comment by sporto** [comment]
So *Enquiry tells sequel to try to expand this as an array will do, we end up passing all records in the DB

**Comment by nathanl** [comment]
Sebastian - I definitely consider this a bug. Thanks for reporting it. :smile: One of us will get to fixing it sooner or later, unless you decide to submit a PR yourself.

**Comment by nathanl** [comment]
Your little Rake task example is great, by the way, and since you said "`*Enquiry` tells sequel to try to expand this as an array will do", I thought, "I wonder if `*thing` is syntactic sugar for `thing.to_a`." Maybe you already knew this, but it is.

```ruby
class Example
  def to_a
    ["woo a string in an array"]
  end
end

e = Example.new
o = *e # => ["woo a string in an array"]

**Comment by sporto** [comment]
Right, I see, they do defined to `to_a`, so it behaves like an array.

**Comment by nathanl** [comment]
@sporto I think I have this fixed. Would you try the `fix_sequel_bug` branch and verify that it works for you? https://github.com/nathanl/authority/pull/105

**Comment by sporto** [comment]
@nathanl 	Yes, this works!

**Comment by nathanl** [comment]
Awesome! Merging.

**Comment by nathanl** [comment]
Fix released in 3.1.0
